src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -141,27 +142,31 @@ public File getLogFile() {
     public void start(AbstractProject<?,?> project, boolean newInstance) {
         super.start(project, newInstance);
         if (newInstance && getDescriptor().isManageHook()) {
-            // make sure we have hooks installed. do this lazily to avoid blocking the UI thread.
-            final Collection<GitHubRepositoryName> names = GitHubRepositoryNameContributor.parseAssociatedNames(job);
-
-            getDescriptor().queue.execute(new Runnable() {
-                public void run() {
-                    LOGGER.log(Level.INFO, "Adding GitHub webhooks for {0}", names);
-                    OUTER:
-                    for (GitHubRepositoryName name : names) {
-                        for (GHRepository repo : name.resolve()) {
-                            try {
-                                if(createJenkinsHook(repo, getDescriptor().getHookUrl())) {
-                                    continue OUTER;
-                                }
-                            } catch (Throwable e) {
-                                LOGGER.log(Level.WARNING, "Failed to add GitHub webhook for "+name, e);
+            registerHooks();
+        }
+    }
+
+    public void registerHooks() {
```

Comment:
Commit description says "Move to separate method to have ability re-register hooks" from groovy for example.